packages/core/src/agent/MessageSender.ts

```diff
@@ -327,7 +333,11 @@ export class MessageSender {
       }
 
       const encryptedMessage = await this.envelopeService.packMessage(agentContext, message, keys)
-      await this.messageRepository.add(connection.id, encryptedMessage)
+      await this.messagePickupRepository.addMessage({
+        connectionId: connection.id,
+        recipientKeys: keys.recipientKeys.map((item) => verkeyToDidKey(item.publicKeyBase58)),
```

Comment:
I'd prefer:

```suggestion
        recipientKeys: keys.recipientKeys.map((item) => new DidKey(item).did),
```

However, is there a need to transform the key from a key instance to a `did:key` instance? The longer we can keep it a `Key` instance the better TS can help us and it's easier to understand. Because string can be `did:key:xxx` or just a plain base58 encoded public key.

So my preference would be to pass an instance of `Key` to the `messagePickupRepository`, or rename `recipientKeys` to something more unique so it's clear how this `recipientKey` is encoded